torch/_guards.py

```diff
@@ -53,12 +53,13 @@
 
 
 class CompileId(NamedTuple):
-    frame_id: int
+    compiled_autograd_id: Optional[int]
+    frame_id: Optional[int]
     # This id is per-frame, and counts how many times we've compiled this
     # frame.  This could have been a global id but having this be per-frame
     # gives you a better intuitive sense for how many recompiles have occurred
     # so far.
-    frame_compile_id: int
+    frame_compile_id: Optional[int]
```

Comment:
@ezyang I'm thinking of marking everything in this class as optional, do you have any concerns? They'll need to be optional in the tlparse repo too

Reply:
Beh, it's not clear this can be a NamedTuple anymore, you are reordering the field numbers

Reply:
Not sure if the NamedTuple was load bearing? If it wasn't we can just make this a dataclass and the compiled_autograd_id is default None

Reply:
no instances of indexing, i changed it to dataclass. we'll move to kw_only when 3.9 is dropped